### PR TITLE
(maint) Use Bolt::PAL for interactions with puppet

### PR DESCRIPTION
**Pull request by adreyer**
Previously the PAL class was only used for tests and was duplicated in
CLI. This converts CLI to interact with Puppet through the PAL clas
instead.

### CODE REVIEW

File: lib/bolt/pal.rb [hunk 1]
```diff
@@ -1,15 +1,27 @@
...
     BOLTLIB_PATH = File.join(__FILE__, '../../../modules')
 
     def initialize(config)
-      # TODO: how should we manage state? Does noop go here?
+      # Nothing works without initialized this global state. Reinitializing
+      # is safe and in practice only happen in tests
+      self.class.load_puppet
+      self.class.configure_logging(config[:log_level])
+
+      # TODO: how should we manage executor state?
```
Comment (MikaelSmith) [inline lib/bolt/pal.rb]: Not sure what this means.

Comment (MikaelSmith) [reply lib/bolt/pal.rb]: Why not pass the executor in to run_task/run_plan? Seems as straight-forward as passing `noop` through.

File: spec/spec_helper.rb [hunk 2]
```diff
@@ -4,6 +4,7 @@
 require 'bolt/logger'
 require 'logging'
 require 'rspec/logging_helper'
+require_relative '../vendored/require_vendored'
```
Comment (MikaelSmith) [inline spec/spec_helper.rb]: Why is this needed? Doesn't load_puppet for particular tests cover it?

Comment (MikaelSmith) [reply spec/spec_helper.rb]: Or... why did this work before?

Comment (adreyer) [reply spec/spec_helper.rb]: we may try to reset puppet's settings before any test has called load_puppet.

Comment (adreyer) [reply spec/spec_helper.rb]: I'm not sure how it worked before

File: lib/bolt/cli.rb [hunk 3]
```diff
@@ -394,44 +395,33 @@ def handle_parser_errors
...
 
       if options[:mode] == 'plan'
         executor = Bolt::Executor.new(@config, options[:noop], true)
-        execute_plan(executor, options)
+        result = pal.run_plan(options[:object], options[:task_options], executor)
+        outputter.print_plan(result)
+        # An exception would have been raised if the plan failed
         code = 0
       else
+        # This executor isn't used for tasks
```
Comment (MikaelSmith) [inline lib/bolt/cli.rb]: Comment is no longer correct.

File: spec/bolt/pal_spec.rb [hunk 4]
```diff
@@ -0,0 +1,2 @@
+# Currently PAL is tested through the CLI execute method. New tests should be
+# added here as we add features  or fix bugs.
```
Comment (MikaelSmith) [inline spec/bolt/pal_spec.rb]: Extra space you could remove.

**Comment by MikaelSmith** [comment]
Needs a rebase now.
